### ISSUE TITLE
Improper error handling in matchAgainstPrefix()

### ISSUE THREAD

**Issue by KyleMaas** [ssb-ngi-pointer/jitdb#92]
Getting the following error something in ssb-browser-demo.  Not sure how I triggered it, but it is listed as a FIXME, so I figured I'd better report that it's being hit out in the wild.

```
Uncaught (in promise) TypeError: Cannot destructure property 'value' of 'recs[i]' as it is undefined.
    at bundle-ui.js:73367
    at bundle-ui.js:90208
    at matchAgainstPrefix (bundle-ui.js:73362)
    at bundle-ui.js:73400
    at ensureIndexSync (bundle-ui.js:73298)
    at getBitsetForOperation (bundle-ui.js:73399)
    at bundle-ui.js:73429
    at bundle-ui.js:73333
    at ensureIndexSync (bundle-ui.js:73298)
    at getFullBitset (bundle-ui.js:73329)
```

https://github.com/ssb-ngi-pointer/jitdb/blob/811cba90652ff13052fac0613550b33f58423784/index.js#L594

**Comment by arj03** [comment]
Do you by any chance know if you where doing deletes before this happened?

**Comment by staltz** [comment]
Should be fixed by #90, let us know if that's not the case. :)

**Comment by KyleMaas** [comment]
@arj03 I don't believe any deletion happened prior to this.  I was work on the "deleting messages when blocking a user" feature of ssb-browser-demo near that time, but I did that in a different private window which wouldn't have had an effect on the instance that was exhibiting errors.

@staltz Appears to have fixed the problem.  Thanks!
